lib/at_host/at_host.c

```diff
@@ -70,49 +70,25 @@ static inline void write_uart_string(const char *str)
 	}
 }
 
-static void response_handler(void *context, const char *response)
+static void response_handler(const char *response)
 {
-	ARG_UNUSED(context);
-
 	/* Forward the data over UART */
 	write_uart_string(response);
 }
 
 static void cmd_send(struct k_work *work)
 {
-	char              str[25];
-	enum at_cmd_state state;
 	int               err;
 
 	ARG_UNUSED(work);
 
-	err = at_cmd_write(at_buf, at_buf,
-			   sizeof(at_buf), &state);
+	err = nrf_modem_at_cmd(at_buf, sizeof(at_buf), "%s", at_buf);
 	if (err < 0) {
 		LOG_ERR("Error while processing AT command: %d", err);
-		state = AT_CMD_ERROR;
 	}
 
-	/* Handle the various error responses from modem */
-	switch (state) {
-	case AT_CMD_OK:
-		write_uart_string(at_buf);
-		write_uart_string(OK_STR);
```

Comment:
I guess you could remove `OK_STR` and `ERROR_STR` definitions from the top of the file.

Reply:
Also check whether `INVALID_DESCRIPTOR` is used.